cmd/sriov/main.go

```diff
@@ -102,12 +101,21 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}}
 
 	if !netConf.DPDKMode {
-		macAddr, err = sm.SetupVF(netConf, args.IfName, netns)
+		err = sm.SetupVF(netConf, args.IfName, netns)
 
 		if err != nil {
 			return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, netConf.Master, err)
 		}
-		result.Interfaces[0].Mac = macAddr
+	}
+
+	// add the mac address we report to the CNI call return object
```

Comment:
I don't think we need to introduce the new variables CurrentEffectiveMAC and CurrentAdminMAC. We can keep this simpler. Note that when conf.Mac is set, this value is configured as effective and admin Mac, so regardless of the type of driver, we can simply say.

	if netConf.MAC != "" {
		result.Interfaces[0].Mac = netConf.MAC
	} else {
		if !netConf.DPDKMode {
			result.Interfaces[0].Mac = netConf.OrigVfState.EffectiveMAC
		} else {
			result.Interfaces[0].Mac = netConf.OrigVfState.AdminMAC
		}
	}

For this to work, the only changed you need is to move line 85 to line 81 (OrigVfState.EffectiveMAC). Note that the AdminMAC is always obtained. I think it's easier to understand if we explicitly mention where the mac is coming from and having less variables. 

My only doubt is about mac "00:00:00:00:00:00", why don't you want to show it? If it is the one set in the VF then it should be presented.

Reply:
I love the idea I fully miss that solution it's much cleaner thanks!!!

and I will still leave the 00 mac address out.
we inject the network-status into the pod as a json file and I will like the customer to check if the mac is "" meaning he can use whatever mac he wants in the application.
I think is better than reporting 00 on the mac address.